workflow/validate/validate.go

```diff
@@ -1374,7 +1379,7 @@ func (ctx *templateValidationCtx) validateDAG(scope map[string]interface{}, tmpl
 		ancestry := common.GetTaskAncestry(dagValidationCtx, task.Name)
 		for _, ancestor := range ancestry {
 			ancestorTask := dagValidationCtx.GetTask(ancestor)
-			resolvedTmpl := resolvedTemplates[ancestor]
+			resolvedTmpl = resolvedTemplates[ancestor]
```

Comment:
This change doesn't actually do anything but is probably a wrong change and at minimum would be best reverted.

`resolvedTmpl` is not used outside of this for loop in the outer scope **after** this loop. So the fact that this variable used to shadow the outer `resolvedTmpl` but doesn't any more doesn't actually change the behaviour. However - this value of resolvedTmpl is definitely _not_ useful outside of this loop, and the formerly shadowed version might be. The right fix is to use a different name for this variable inside this loop.

Reply:
reverting this